File: csharp/src/Repl/Repl.cs
public static class Repl
{
    private const string PROMPT = ">> ";

    public static void Start()
    {
        while (true)
        {
            Console.Write(PROMPT);
            var line = Console.ReadLine();
            
            if (string.IsNullOrEmpty(line))
                return;

            var lexer = new Lexer(line);
            var token = lexer.NextToken();

            while (token.Type != TokenType.Eof)
            {
                Console.WriteLine($"Type: {token.Type}, Literal: {token.Literal}");
                token = lexer.NextToken();
            }
        }
    }
} 

File: rust/src/repl.rs
use crate::lexer::{Lexer, TokenType};
use std::io::{self, BufRead};

const PROMPT: &str = ">>";

pub fn start_repl() {
    loop { 
        let mut buffer = String::new();
        io::stdin().read_line(&mut buffer).unwrap();
        
        let mut lex = Lexer::new(buffer);
        let mut tok = lex.next_token();

        while tok.Type != TokenType::Eof {
            println!("Token literal {}", &tok.Literal);
            tok = lex.next_token();
        }
    }
}
